src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -110,5 +108,59 @@ internal static bool CompletionRequiresQuotes(string completion, bool escape)
 
         private static bool ContainsCharsToCheck(ReadOnlySpan<char> text, bool escape)
             => text.ContainsAny(escape ? s_escapeCharsToCheck : s_defaultCharsToCheck);
+
+        /// <summary>
+        /// Quotes and escapes a given completion text based on the specified options.
+        /// </summary>
+        /// <param name="completionText">
+        /// The text to be quoted and potentially escaped.
+        /// </param>
+        /// <param name="quote">
+        /// The quote character to use for enclosing the text. Defaults to a single quote ('') if not provided.
+        /// </param>
+        /// <param name="escapeSingleQuoteChars">
+        /// Indicates whether single quote characters in the text should be escaped. Defaults to <c>true</c>.
+        /// </param>
+        /// <param name="escapeDoubleQuoteChars">
+        /// Indicates whether special characters (e.g., backticks and dollar signs) in the text
```

Comment:
Please reword without "e.g.".
I think for good docs it is better to have exact list of the chars here or in remarks section.